### PR TITLE
update save-load-unix-test

### PR DESCRIPTION
**Pull request by xlgao-zju**
A part of #16756

update `integration-cli/docker_cli_save_load_unix_test.go`

Signed-off-by: xlgao-zju xlgao@zju.edu.cn

### CODE REVIEW

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 1]
```diff
@@ -41,41 +41,31 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
...
 	cmd.Stdin = tty
 	cmd.Stdout = tty
 	cmd.Stderr = tty
-	if err := cmd.Start(); err != nil {
-		c.Fatalf("start err: %v", err)
-	}
-	if err := cmd.Wait(); err == nil {
-		c.Fatal("did not break writing to a TTY")
-	}
+	c.Assert(cmd.Start(), check.NotNil)
```
Comment (HuKeping) [inline integration-cli/docker_cli_save_load_unix_test.go]: It should be assert as `IsNil`

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 2]
```diff
@@ -41,41 +41,31 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
...
+	c.Assert(
```
Comment (HuKeping) [inline integration-cli/docker_cli_save_load_unix_test.go]: I think one line for this is OK

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 3]
```diff
@@ -41,41 +41,31 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
...
 
 	n, err := pty.Read(buf)
-	if err != nil {
-		c.Fatal("could not read tty output")
-	}
-
-	if !bytes.Contains(buf[:n], []byte("Cowardly refusing")) {
-		c.Fatal("help output is not being yielded", out)
-	}
+	c.Assert(err, check.NotNil, check.Commentf("could not read tty output"))
```
Comment (HuKeping) [inline integration-cli/docker_cli_save_load_unix_test.go]: The same, shouldn't be `NotNil`

**Comment by xlgao-zju** [comment]
@HuKeping thx, I just misunderstood the usage of `c.Assert`.

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 4]
```diff
@@ -29,9 +29,8 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
 	saveCmd := exec.Command(dockerBinary, "save", repoName)
 	saveCmd.Stdout = tmpFile
 
-	if _, err = runCommand(saveCmd); err != nil {
-		c.Fatalf("failed to save repo: %v", err)
-	}
+	_, err = runCommand(saveCmd)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_save_load_unix_test.go]: Can we use `dockerCmd` instead? It checks for errors and gives a nicely formatted error message on failure.

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 5]
```diff
@@ -41,41 +40,27 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
 	loadCmd := exec.Command(dockerBinary, "load")
 	loadCmd.Stdin = tmpFile
 
-	if out, _, err = runCommandWithOutput(loadCmd); err != nil {
-		c.Fatalf("failed to load repo: %s, %v", out, err)
-	}
+	out, _, err = runCommandWithOutput(loadCmd)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_save_load_unix_test.go]: `dockerCmd`

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 6]
```diff
@@ -41,41 +40,27 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
...
+	c.Assert(err, check.IsNil, check.Commentf(out))
 
 	after, _ := dockerCmd(c, "inspect", repoName)
 
-	if before != after {
-		c.Fatalf("inspect is not the same after a save / load")
-	}
+	c.Assert(before != after, check.Equals, false, check.Commentf("inspect is not the same after a save / load"))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_save_load_unix_test.go]: `c.Assert(before, check.Not(check.Equals), after)`

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 7]
```diff
@@ -41,41 +40,27 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
...
 	n, err := pty.Read(buf)
-	if err != nil {
-		c.Fatal("could not read tty output")
-	}
-
-	if !bytes.Contains(buf[:n], []byte("Cowardly refusing")) {
-		c.Fatal("help output is not being yielded", out)
-	}
+	c.Assert(err, check.IsNil, check.Commentf("could not read tty output"))
+	c.Assert(!bytes.Contains(buf[:n], []byte("Cowardly refusing")), check.Equals, false, check.Commentf("help output is not being yielded", out))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_save_load_unix_test.go]: `c.Assert(string(buf[:n]), checker.Contains, "Cowardly refusing")`

File: integration-cli/docker_cli_save_load_unix_test.go (hunk 4 above)
Comment (xlgao-zju) [inline integration-cli/docker_cli_save_load_unix_test.go]: Oh, that is great!

Comment (vdemeester) [inline integration-cli/docker_cli_save_load_unix_test.go]: @cpuguy83 the thing is that `dockerCmd` doesn't let you setup the `Stdout` of the command (which is done there).

File: integration-cli/docker_cli_save_load_unix_test.go (hunk 5 above)
Comment (vdemeester) [inline integration-cli/docker_cli_save_load_unix_test.go]: Same as above :sweat_smile:

File: integration-cli/docker_cli_save_load_unix_test.go (hunk 4 above)
Comment (cpuguy83) [inline integration-cli/docker_cli_save_load_unix_test.go]: Oh yep, sorry, overlooked that we need to save to the tmpfile.

Comment (vdemeester) [inline integration-cli/docker_cli_save_load_unix_test.go]: That would be cool if it did though.. :stuck_out_tongue_closed_eyes:

**Comment by xlgao-zju** [comment]
@cpuguy83 @vdemeester  code is updated, plz review.

**Comment by vdemeester** [comment]
@xlgao-zju Failure on the tests :sweat_smile: …

**Comment by xlgao-zju** [comment]
@vdemeester  Sorry, I didn't check the note...I used that note directly...I'll get it right.

**Comment by vdemeester** [comment]
@xlgao-zju Could you squash your commits ? (other than that, looks good :wink:)

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 8]
```diff
@@ -41,41 +40,27 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
...
-	}
+	out, _, err = runCommandWithOutput(loadCmd)
+	c.Assert(err, check.IsNil, check.Commentf(out))
 
 	after, _ := dockerCmd(c, "inspect", repoName)
 
-	if before != after {
-		c.Fatalf("inspect is not the same after a save / load")
-	}
+	c.Assert(before, check.Equals, after, check.Commentf("inspect is not the same after a save / load"))
```
Comment (MHBauer) [inline integration-cli/docker_cli_save_load_unix_test.go]: this commentf can be a standard comment on the same line as there are no variables being interpolated.
`// inspect is not the same after a save / load`

File: integration-cli/docker_cli_save_load_unix_test.go [hunk 9]
```diff
@@ -41,41 +40,27 @@ func (s *DockerSuite) TestSaveAndLoadRepoStdout(c *check.C) {
...
 	cmd.Stdout = tty
 	cmd.Stderr = tty
-	if err := cmd.Start(); err != nil {
-		c.Fatalf("start err: %v", err)
-	}
-	if err := cmd.Wait(); err == nil {
-		c.Fatal("did not break writing to a TTY")
-	}
+	c.Assert(cmd.Start(), check.IsNil)
+	c.Assert(cmd.Wait(), check.NotNil, check.Commentf("did not break writing to a TTY"))
```
Comment (MHBauer) [inline integration-cli/docker_cli_save_load_unix_test.go]: same
this commentf can be a standard comment on the same line as there are no variables being interpolated.

**Comment by xlgao-zju** [comment]
@vdemeester updated.

**Comment by cpuguy83** [comment]
LGTM

**Comment by vdemeester** [comment]
LGTM 🐯

**Pull request merged**